### ISSUE TITLE
Refactor Api to be upto date

### ISSUE THREAD

**Issue by jantimon** [evilebottnawi/favicons#186]
Hey @evilebottnawi

we are right now refactoring the [webpack-favicons-plugin](https://github.com/jantimon/favicons-webpack-plugin) which is a heavy user of your plugin (~40.000 downloads per month)

@milewski is now thinking about forking / rewriting your favicons plugin because he believes that we can't implement the requirements for modern progressive webapps without a lot of wrapping code:

https://github.com/jantimon/favicons-webpack-plugin/issues/88#issuecomment-321443443

I would be very happy if you would join our efforts to improve the current state of favicons for progressive web apps.  
Do you think those changes could be a new version of evilebottnawi/favicons or should they move to a new project?

Thanks :)

**Comment by evilebottnawi** [comment]
@jantimon It's great, now I do not have time to help :disappointed: But I add your to the contributor and PR welcome.

**Comment by jantimon** [comment]
Okay can understand that - thanks I'll talk to @milewski 👍

**Comment by brunocodutra** [comment]
@milewski you seem to have a very good understanding of the current state of PWAs, could you summarize what exactly needs to be done on our side?
